okio/jvm/src/main/java/okio/Options.kt

```diff
@@ -31,7 +32,209 @@ class Options private constructor(
   companion object {
     @JvmStatic
     fun of(vararg byteStrings: ByteString): Options {
-      return Options(byteStrings.clone()) // Defensive copy.
+      // Sort the byte strings which is required when to recursively build the trie. Map the sorted
+      // indexes to the caller's indexes.
+      val sortedByteStrings = byteStrings.sortedArray()
+      // TODO(jwilson): this is a O(N*N) loop to recover the original indexes. Ick.
+      val sortedIndexes = IntArray(sortedByteStrings.size) {
+        byteStrings.indexOf(sortedByteStrings[it])
+      }
+
+      require(sortedByteStrings.isNotEmpty()) { "no options provided" }
+      require(sortedByteStrings[0].size() > 0) { "the empty byte string is not a supported option" }
```

Comment:
Both of these requirements are new as of this change. Dropping support for them makes the code simpler.